### DISCUSSION TITLE
Consideration of Refresh Token Rotation?

### DISCUSSION THREAD

**Discussion by turboturtleD** [etsy/open-api#848] [Q&A]
I was shocked to discover that I have been inadvertently using the same refresh token over and over (thousands of times, no doubt) to obtain new access tokens over the course of weeks and months.  I would have assumed that when a refresh token was exchanged, and a new refresh token issued upon use, that the prior token would be invalidated.  I guess I had assumed (seemingly incorrectly) that Refresh Token Rotation was in play as laid out on the auth0.com blog:

https://auth0.com/blog/refresh-tokens-what-are-they-and-when-to-use-them/

Seems maybe it's common practice to not do this, but maybe best practice?  Just seemed like a no brainer (to implement refresh token rotation on the Auth server).

**Comment by Gareth064** [comment]
Sorry are you saying that you haven't been given a new refresh token after using the previous one. Or you just kept using the previous one (even tho a new one was given) and it continued to work?

**Reply by turboturtleD** [reply]
I've received thousands of new refresh tokens (_and used none of them_), one for each [successful] call for the past several months.

The whole time, I was inadvertently just using the same original refresh token over and over and over, and it has continued to work for months (due to expire soon - though the bug that perpetuated this in my code has been fixed).

**Comment by Gareth064** [comment]
Gotcha.

And the idea of course is that if a refresh token is used, and a new one is provided, then the previous should be invalid.

Definitely a bad bug there in that case.


Kind Regards

Gareth Doherty



It’s nice to be important,

But it’s more important to be nice.



________________________________
